### PR TITLE
feat: add batch size to batch write

### PR DESCRIPTION
**Pull request by fqutishat**
Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

**Review by DRK3 — APPROVED** [review]

**Review by bstasyszyn — APPROVED** [review]

File: pkg/storage/formattedstore/batchwrite.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ type BatchService struct {
 	formatter        Formatter
 	provider         batchProvider // TODO remove batch provider after refactoring edv batch
 	current          *pendingBatch
+	batchSizeLimit   int
```
Comment (troyronda) [inline pkg/storage/formattedstore/batchwrite.go:38]: We could also additionally consider an optional max batch time in a followup.

File: pkg/storage/formattedstore/batchwrite.go [hunk 2]
```diff
@@ -148,10 +164,13 @@ func newPendingBatch(provider batchProvider, formatter Formatter) *pendingBatch
 	return &pendingBatch{
 		provider:  provider,
 		formatter: formatter,
+		size:      0,
```
Comment (troyronda) [inline pkg/storage/formattedstore/batchwrite.go:167]: Does 0 mean no maximum or disable batch?

Should also have a way to indicate no maximum.

**Review by troyronda — APPROVED** [review]

File: pkg/storage/formattedstore/batchwrite.go (hunk 1 above)
Comment (fqutishat) [reply pkg/storage/formattedstore/batchwrite.go:38]: yes

File: pkg/storage/formattedstore/batchwrite.go (hunk 2 above)
Comment (fqutishat) [reply pkg/storage/formattedstore/batchwrite.go:167]: I will add case for no maximum

Comment (fqutishat) [reply pkg/storage/formattedstore/batchwrite.go:167]: done

**Pull request merged**